### PR TITLE
bin: add scan command

### PR DESCRIPTION
**Pull request by BusyJay**
@ngaut @siddontang @disksing PTAL

### CODE REVIEW

**Comment by siddontang** [comment]
Add a test for escape/unesacpe?

**Comment by BusyJay** [comment]
They are tested by doc test.

**Comment by siddontang** [comment]
No, what I mean is first escape the string, and then unescape, then check with origin.

**Comment by BusyJay** [comment]
What's the difference with current doc tests' cases?

**Comment by siddontang** [comment]
LGTM

File: src/bin/tikv-dump.rs [hunk 1]
```diff
@@ -53,7 +53,13 @@ fn main() {
...
     opts.optflag("", "info", "print the region info");
     opts.optopt("i", "index", "set the raft log index", "");
-    opts.optopt("k", "key", "set the query raw key, in hex format", "");
+    opts.optopt("k", "key", "set the query raw key, in escape format", "");
+    opts.optopt("f",
+                "from",
+                "set the scan from raw key, in escaped format",
+                "");
+    opts.optopt("t", "to", "set the scan end raw key, in escaped format", "");
+    opts.optopt("l", "limit", "set the scan limit kesy", "");
```
Comment (disksing) [inline src/bin/tikv-dump.rs]: "set the scan limit"?

File: src/util/mod.rs [hunk 2]
```diff
@@ -156,17 +156,20 @@ pub fn to_socket_addr<A: ToSocketAddrs>(addr: A) -> io::Result<SocketAddr> {
...
 pub fn escape(data: &[u8]) -> String {
     let mut escaped = Vec::with_capacity(data.len() * 4);
     for &c in data {
         match c {
-            b'\'' => escaped.extend_from_slice(b"\\'"),
+            b'\n' => escaped.extend_from_slice(br"\n"),
+            b'\r' => escaped.extend_from_slice(br"\r"),
+            b'\t' => escaped.extend_from_slice(br"\t"),
             b'"' => escaped.extend_from_slice(b"\\\""),
+            b'\\' => escaped.extend_from_slice(br"\\"),
```
Comment (disksing) [inline src/util/mod.rs]: Missing `'`? 
Here's an implementation from pb https://github.com/google/protobuf/blob/9240acd179708c3a2ab79508af6fd68efa6f883a/src/google/protobuf/stubs/strutil.cc#L482

**Comment by disksing** [comment]
LGTM

**Pull request merged**